### PR TITLE
Allow interpolateColor()'s outputRange to contain numbers

### PR DESCRIPTION
**Pull request by wcandillon**

### CODE REVIEW

File: react-native-reanimated.d.ts [hunk 1]
```diff
@@ -455,7 +455,7 @@ declare module 'react-native-reanimated' {
     export function interpolateColor(
       value: number,
       inputRange: number[],
-      outputRange: string[],
+      outputRange: (string | number)[],
```
Comment (jakub-gonet) [inline react-native-reanimated.d.ts:458]: We can also add `readonly` modifier here, we don't mutate those arrays.

**Review by jakub-gonet — APPROVED** [review]
LGTM, thanks!

**Comment by jakub-gonet** [comment]
One more thing, could you add some tests in the TS file?

**Comment by mrousavy** [comment]
The `interpolateColor()` function returns `string | number`, but a View Style doesn't allow `number` for colors such as `backgroundColor`, did you also notice this? That means, this code:

```ts
const x = useAnimatedStyle(() => ({ backgroundColor: interpolateColor(...) }), []); 
```

will raise a `ts-error` since `number` is not assignable to `backgroundColor`.

**Comment by wcandillon** [comment]
@mrousavy This is fixed in #1531 ?

**Comment by mrousavy** [comment]
@wcandillon no, since `ColorValue` is defined as `string | unique symbol`.

**Comment by wcandillon** [comment]
Using #1531 as a patch, I'm able to return `const x = useAnimatedStyle(() => ({ backgroundColor: interpolateColor(...) }), []);`

**Comment by mrousavy** [comment]
@wcandillon oh, you're right I missed the conditional `extends`, my bad! Works for me 👍

**Comment by wcandillon** [comment]
Let's merge it ;-)

**Comment by jakub-gonet** [comment]
@wcandillon, could you add some tests to the TS test file too?

**Comment by wcandillon** [comment]
@jakub-gonet done, CI should pass.

**Comment by wcandillon** [comment]
@jakub-gonet the CI failure seems unrelated, is that possible?

**Comment by jakub-gonet** [comment]
Sorry for the delay, let's trigger CI and see if it's working now.

**Pull request merged**